### PR TITLE
Including metrics on number of batches handled through persistent queue

### PR DESCRIPTION
**Pull request by pmm-sumo**
Testing: Unit tests updated, manual tests to follow

### CODE REVIEW

File: exporter/exporterhelper/internal/persistent_storage.go [hunk 1]
```diff
@@ -360,6 +360,8 @@ func (pcs *persistentContiguousStorage) itemDispatchingStart(ctx context.Context
 		pcs.logger.Debug("Failed updating currently dispatched items",
 			zap.String(zapQueueNameKey, pcs.queueName), zap.Error(err))
 	}
+
+	recordCurrentlyDispatchedBatches(ctx, len(pcs.currentlyDispatchedItems), pcs.queueName)
```
Comment (tigrannajaryan) [inline exporter/exporterhelper/internal/persistent_storage.go:364]: Do we only record these new metrics for persistent queue or also for in-memory queue when persistent queue is disabled?
I think they are useful regardless of whether persistence is enabled or no.

**Review by tigrannajaryan — APPROVED** [review]
LGTM, just one question.

**Review by bogdandrutu — CHANGES_REQUESTED** [review]
@tigrannajaryan I am not sold on this, first we should stop adding "random" metrics (we need a plan of how to name metrics, what is important vs not important to monitor, etc.) because sooner than later collector's own telemetry will need to stabilize, and so far I am the only one dealing with this work.

File: exporter/exporterhelper/internal/persistent_storage.go (hunk 1 above)
Comment (pmm-sumo) [reply exporter/exporterhelper/internal/persistent_storage.go:364]: Yeah, I was wondering about doing that for in-memory queue as well though it seems it's using the old API and metrics would need to be pretty much redone there

**Comment by pmm-sumo** [comment]
> I am not sold on this, first we should stop adding "random" metrics (we need a plan of how to name metrics, what is important vs not important to monitor, etc.) because sooner than later collector's own telemetry will need to stabilize, and so far I am the only one dealing with this work.

That's a fair argument. I think `queued_retry` might use redoing metrics. If you have some suggestion @bogdandrutu how would you like to see them organised I will be happy to take care of that. 

BTW, I was thinking about adding some more metrics in the future to persistent queue which would hold info on number of operations underneath

**Comment by tigrannajaryan** [comment]
> @tigrannajaryan I am not sold on this, first we should stop adding "random" metrics (we need a plan of how to name metrics, what is important vs not important to monitor, etc.) because sooner than later collector's own telemetry will need to stabilize, and so far I am the only one dealing with this work.

Sounds good. Perhaps that can also eventually feed into a broader scope of defining semantic conventions for "data processors" in Otel spec, not just the Collector.

**Comment by pmm-sumo** [comment]
Considering the discussion above and idea of leveraging a better strategy for collector metrics overall, I am closing this PR

**Pull request closed without merging**